haec-user-service/src/modules/user/services/user.service.ts

```diff
@@ -46,14 +49,47 @@ export class UserService {
           email: input.email,
         },
       });
-
       if (await bcrypt.compare(input.password, user.hashedPass)) {
         return true;
       }
     } catch (error) {
       console.error(error);
     }
-
     return false;
   }
+
+// This will log the user ** IN **
+
+
+
+  public async loginUser(input: {
+    email: string;
+    password: string;
+  }): Promise<string> {
+    const verified = await this.verifyUser({ email: input.email, password: input.password})
+    if (verified == true){
+      // Generates random 21 character token with Crypto
+      const token = this.randomString()
+      // debug
+      console.log(token)
+      await this.prisma.user.update({
+        where: { email: input.email },
+        // assigning `token variable` throws error that it `cannot accept a Promise value`
+        data: { token: "token"} })
+      return token
```

Comment:
Working on accessing the `return value` from `randomString()`, but because it returns a `Promise { }`, I can't use it on line 78. 
`Console.log(token)` --> `Promise { '9XfM90X5g/nu/rNkjjWT0' }`, but I just want the contents of the Promise. Point me in the right direction please?

Reply:
I'm reading `https://dev.to/ramonak/javascript-how-to-access-the-return-value-of-a-promise-object-1bck` and she seems to have a good understanding of Promises, and she says `"You can use the return value from the Promise object ONLY in a async function. You should do all the stuff that you'd like to do with the return result in a [randomString ()] async function."`